### ISSUE TITLE
Plunger map high value too high

### ISSUE THREAD

**Issue by phoenixdigital** [Jerware/PinSim#2]
I've just finished my pinsim and was having major issues with the plunger. After calibration whenever I pulled the plunger to the full extent it jumped to zero when viewing in windows testing app 

See Y Rotation in example below. When plunger is at full extent this drops to the far left.
http://i.imgur.com/cC2MGZO.png

I resolved it by changing the lines 

```
    if (currentlyPlunging)
    {
      controller.stickUpdate(STICK_RIGHT, 0, map(distanceBuffer, plungerMaxDistance, plungerMinDistance, zeroValue, -32760));
    }
    else controller.stickUpdate(STICK_RIGHT, 0, map(distanceBuffer, plungerMaxDistance, plungerMinDistance, 0, -32760));

```

I suspect the issue is the -32768 value was somehow too high something to do with masking but the lower number resolved it. I probably could have gotten away with -32767 but I was so stoked to have fixed it I just left it in place.

Thanks again for the code it all works fantastic now.

**Comment by Jerware** [comment]
Weird! Did you experience the problem with my compiled code, or did you compile your own? Odd that I wouldn't have seen this, but nice fix! I'm glad you were able to solve it.

**Comment by phoenixdigital** [comment]
I did compile myself as I added code in to expand the functionality of your button mapping as well as added code to handle a coin input for the MAME/Arcade component of the pinsim.

 It very well could be the compiler as it appears the Right Stick uses a 16 bit number ranging from 

right analog fully up = 32767 = 0x7FFF = 01111111 11111111
to
right analog fully down = -32768 = 0x8000 = 10000000 00000000

Your code should work as is but for some reason the conversion of -32768 gets interpreted as right analog near full up position. It works now with my code tweak so no biggie. Hopefully if someone else encounters this it helps them find a solution.

My versions 

- OS: Fedora 30 (5.2.8-200.fc30.x86_64)
- Arduino: 1.8.2
- Teensyduino: 1.46

UNRELATED: I enhanced your idea of the using the back button combined with other buttons to double the potential number of keys I could map in MAME for the arcade side of things. I changed BACK to a kind of shift key which I used to also have access to left and right stick click (BUTTON_L3 & BUTTON_R3)

Pics of build here - https://imgur.com/a/hNi8sYX

My code changes here if your interested - https://pastebin.com/XCNcJKX9
